main.sh

```diff
@@ -109,25 +125,42 @@ while read line; do
   fi
 done <<< "${usage}"
 
+# Allow long options like --this
+opts="${opts}-:"
+
 # Reset in case getopts has been used previously in the shell.
 OPTIND=1
 
 # Overwrite $arg_<flag> defaults with the actual CLI options
 while getopts "${opts}" opt; do
-  line="$(echo "${usage}" |grep "\-${opt}")"
-
-
-  [ "${opt}" = "?" ] && help "Invalid use of script: ${@} "
-  varname="arg_${opt:0:1}"
-  default="${!varname}"
-
-  value="${OPTARG}"
-  if [ -z "${OPTARG}" ] && [ "${default}" = "0" ]; then
-    value="1"
-  fi
-
-  eval "${varname}=\"${value}\""
-  debug "cli arg ${varname} = ($default) -> ${!varname}"
+  while true; do
+    [ "${opt}" = "?" ] && help "Invalid use of script: ${@} "
+
+    if [ "${opt}" = "-" ]; then #OPTARG is long-option-name or long-option=value
+      if [[ "${OPTARG}" =~ .*=.* ]]; then # --key=value format
+	long=${OPTARG/=*/}
+	eval "opt=\"\${short_opt_${long}}\"" # Set opt to the short option corresponding to the long option
+	OPTARG=${OPTARG#*=}
+      else # --key value format
+        eval "opt=\"\${short_opt_${OPTARG}}\"" # Map long name to short version of option
+	eval "OPTARG=\"\${@:OPTIND:\${has_arg_${opt}}}\"" # Only assign OPTARG if option takes an argument
+	((OPTIND+=has_arg_${opt})) # shift over the argument if argument is expected
+      fi
+      continue # we have set opt/OPTARG to the long value and the argument if it exists
+    else
+      varname="arg_${opt:0:1}"
+      default="${!varname}"
+
+      value="${OPTARG}"
+      if [ -z "${OPTARG}" ] && [ "${default}" = "0" ]; then
+	value="1"
+      fi
+
+      eval "${varname}=\"${value}\""
+      debug "cli arg ${varname} = ($default) -> ${!varname}"
+    fi
+    break
+  done
```

Comment:
This inner `while` loop is a `goto` of sorts; the idea is to detect long options, and then manipulate `opt` `OPTARG` and `OPTIND` to look like the short option version was used. Then go and reprocess it.

I'm about to go to sleep, but now that I look at this hunk, I think we can safely remove this inner while loop. This will also get rid of the extra indentation making this hunk look so big.